salt/modules/aixpkg.py

```diff
@@ -0,0 +1,407 @@
+            info={
+                'changes': ret,
+                'errors': errors
+            }
+        )
+
+    # No error occurred
+    if test:
+        return 'Test succeeded.'
+
+    return ret
+
+
+def remove(name=None, pkgs=None, **kwargs):
+    '''
+    Remove specified fileset(s)/rpm package(s).
+
+    name
+        The name of the fileset or rpm package to be deleted.
+
+
+    Multiple Package Options:
+
+    pkgs
+        A list of filesets and/or rpm packages to delete.
+        Must be passed as a python list. The ``name`` parameter will be
+        ignored if this option is passed.
+
+
+    Returns a list containing the removed packages.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' pkg.remove <fileset/rpm package name>
+        salt '*' pkg.remove tcsh
+        salt '*' pkg.remove xlC.rte
+        salt '*' pkg.remove Firefox.base.adt
+        salt '*' pkg.remove pkgs='["foo", "bar"]'
+    '''
+    targets = salt.utils.args.split_input(pkgs) if pkgs else [name]
+    if not targets:
+        return {}
+
+    if pkgs:
+        log.debug('Removing these fileset(s)/rpm package(s) {0}: {1}'
+            .format(name, targets))
+
+    errors = []
+
+    # Get a list of the currently installed pkgs.
+    old = list_pkgs()
+
+    # Remove the fileset or rpm package(s)
+    for target in targets:
+        try:
+            named, versionpkg, rpmpkg = _check_pkg(target)
+        except CommandExecutionError as exc:
+            if exc.info:
+                errors.append(exc.info['errors'])
+            continue
+
+        if rpmpkg:
+            cmd = ['/usr/bin/rpm', '-e', named]
+            out = __salt__['cmd.run_all'](cmd, output_loglevel='trace')
+        else:
+            cmd = ['/usr/sbin/installp', '-u', named]
+            out = __salt__['cmd.run_all'](cmd, output_loglevel='trace')
+
+    # Get a list of the packages after the uninstall
+    __context__.pop('pkg.list_pkgs', None)
+    new = list_pkgs()
+    ret = salt.utils.data.compare_dicts(old, new)
+
+    if errors:
+        raise CommandExecutionError(
+            'Problems encountered removing filesets(s)/package(s)',
+            info={
+                'changes': ret,
+                'errors': errors
+            }
+        )
+
+    return ret
+
+
+def latest_version(*names, **kwargs):
+    '''
+    Return the latest version of the named fileset/rpm package available for
+    upgrade or installation. If more than one fileset/rpm package name is
+    specified, a dict of name/version pairs is returned.
+
+    If the latest version of a given fileset/rpm package is already installed,
+    an empty string will be returned for that package.
+
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' pkg.latest_version <package name>
+        salt '*' pkg.latest_version <package1> <package2> <package3> ...
+
+    NOTE: As package repositories are not presently supported for AIX
```

Comment:
"package repositories" is redundant as it results in the word "package" appearing twice in the same clause. Also, the 2nd appearance should be "packages", not "package". Thus, this should be:

```
    NOTE: As repositories are not presently supported for AIX installp/RPM
    packages, this function will always return an empty string for a given
    package.
```

Reply:
@terminalmage I hate getting nailed by cut and paste, :(, will fix.